### PR TITLE
Extract wait and findChild in gui tests

### PR DESCRIPTION
**Pull request by eivindjahren**
Create a small utility for getting children of qtwidgets in tests

## Pre review checklist

- [x] Read through the code changes carefully after finishing work
- [x] Make sure tests pass locally (after every commit!)
- [ ] Prepare changes in small commits for more convenient review (optional)
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Updated documentation
- [x] Ensured that unit tests are added for all new behavior (See 
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: tests/unit_tests/gui/conftest.py [hunk 1]
```diff
@@ -437,6 +426,15 @@ def handle_add_dialog():
...
-    QTimer.singleShot(1000, handle_dialog)
-    manage_tool = gui.tools["Manage cases"]
-    manage_tool.trigger()
+    with_manage_tool(gui, qtbot, handle_dialog)
+
+
+V = TypeVar("V")
+
+
+def get_child(gui, typ: Type[V], *args, waiter=None, **kwargs) -> V:
```
Comment (andreas-el) [inline tests/unit_tests/gui/conftest.py:435]: Would it be more accurate to extend name to include the waiting-functionality the qtbot actually does?
It does not just get_child/return child.

Comment (andreas-el) [reply tests/unit_tests/gui/conftest.py:435]: Perhaps add a wait function that uses this function?

Comment (eivindjahren) [reply tests/unit_tests/gui/conftest.py:435]: Something like a separate `wait_for_child` ?

**Review by andreas-el — APPROVED** [review]
Nice one! 🚀

**Pull request merged**